src/wp-includes/user.php

```diff
@@ -682,14 +682,27 @@ function count_many_users_posts( $users, $post_type = 'post', $public_only = fal
 		return $pre;
 	}
 
-	$userlist = implode( ',', array_map( 'absint', $users ) );
-	$where    = get_posts_by_author_sql( $post_type, true, null, $public_only );
+	// Cleanup the users array. Remove duplicates and sort for consistent ordering.
+	$users = array_unique( array_filter( array_map( 'absint', $users ) ) );
```

Comment:
I think `intval` is more appropriate here. We don't want to convert negative integers into user IDs.

```suggestion
	$users = array_unique( array_filter( array_map( 'intval', $users ) ) );
```